File: src/arraysandstrings/IsSubsequence.java
package arraysandstrings;

/**
 * Given two strings s and t, return true if s is a subsequence of t, or false otherwise.
 * Example:
 * For example, "ace" is a subsequence of "abcde" while "aec" is not.
 */

public class IsSubsequence {
    public static boolean isSubsequence(String s, String t) {
        int i = 0, j = 0;

        while (i < s.length() && j < t.length()) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
            }
            j++;
        }

        return i == s.length();
    }

}

File: src/main/kotlin/arraysandstrings/IsSubsequence.kt
package arraysandstrings

class IsSubsequence {
    companion object {

        /**
         * Problem: Given two strings s and t, return true if s is a subsequence of t, or false otherwise.
         * Approach: Two pointers
         *
         * @param s subsequence, example: "ace"
         * @param t string, example: "abcde"
         * @return true if it is subsequence (like "ace"), false if it is not (like "acf")
         */
        fun isSubsequence(s: String, t: String): Boolean {
            var i = 0
            var j = 0

            while (i < s.length && j < t.length) {
                if(s[i] == t[j]) {
                    i++
                }
                j++
            }

            return i == s.length
        }
    }
}